filebeat/input/filestream/input.go

```diff
@@ -166,11 +165,11 @@ func (inp *filestream) Run(
 	})
 	defer streamCancel()
 
-	if err := inp.readFromSource(ctx, log, r, fs.newPath, state, publisher, metrics); err != nil {
-		ctx.UpdateStatus(status.Degraded, fmt.Sprintf("error while reading from source: %v", err))
-		return err
-	}
-	return nil
+	// The only error readFromSource can return is the error from `ctx`,
+	// there is no point in reporting it to the Manager (aka Elastic-Agent).
+	// Also, the caller of Run, will correctly report the error and filter
+	// out 'context cancelled'.
+	return inp.readFromSource(ctx, log, r, fs.newPath, state, publisher, metrics)
```

Comment:
That is the case today, but what if someone changes the logic inside readFromSource to return another kind of error? This promise would then become invalid and could cause silent failures. If you mean that Run up the chain has logic to report the errors then it is fine, then disregard my comment.

Reply:
Thanks, I like your suggestion, it is both clearer and shorter.

Reply:
`UpdateStatus` could probably just unconditionally filter out `context.Cancelled` and the Beats context equivalent. 

`context.Cancelled` is not an actionable user error, it's something that would only ever arise because of a bug. We could log it but not change the agent state perhaps.

Reply:
I agree if the error is `context.Cancelled` it's coming from a bug, however it could also break/stop something essential (like an input) and if that affects the data ingestion or the overall behaviour of the Elastic-Agent it should be reported to the user so they know something is wrong.

Ideally there would be no bug, but if I have to choose between a silent bug and a verbose/noisy one, I'll take the noisy one, it's gonna be much easier to catch and less likely to make it into a final release.